base/linalg/exceptions.jl

```diff
@@ -29,10 +29,29 @@ function Base.showerror(io::IO, ex::ARPACKException)
     end
 end
 
+"""
+   SingularException(info)
+
+The matrix passed is a singular matrix. Optional argument INFO is an INTEGER
+= 0:  successful exit
+< 0:  if INFO = -i, the i-th argument had an illegal value
+> 0:  if INFO = i, U(i,i) is exactly zero. The factorization
+      has been completed, but the factor U is exactly
+      singular, and division by zero will occur if it is used
+      to solve a system of equations.
+"""
 struct SingularException <: Exception
     info::BlasInt
 end
 
+function Base.showerror(io::IO, ex::SingularException)
+    print(io, "SingularException: ")
+    if ex.info > 0
+        print(io,"matrix is singular")
```

Comment:
I'm wondering if we really need a custom show method. In particular when the exception gets a doc string. If we really want a custom show method then we should at least print the value of `info` since it tells you where the factorization failed.

Reply:
I had put in the custom show method because I found one for ARPACKException. SingularException already does print `info` if that is what you mean, but adding clarity regarding the significance of `info` might be helpful I think.

Reply:
With this show method, the `info` field is not shown. The positive value can mean slightly different things depending on the factorization so unless we want a fairly long error message it might be better to refer to the doc string for details.